core/src/main/scala/com/softwaremill/kmq/redelivery/streams/RedeliveryTrackerStream.scala

```diff
@@ -1,37 +1,36 @@
 package com.softwaremill.kmq.redelivery.streams
 
-import akka.Done
 import akka.actor.ActorSystem
 import akka.kafka.ConsumerMessage.CommittableMessage
 import akka.kafka.scaladsl.Consumer
 import akka.kafka.scaladsl.Consumer.DrainingControl
 import akka.kafka.{ConsumerSettings, Subscriptions}
 import akka.stream.ClosedShape
 import akka.stream.scaladsl.{Broadcast, Flow, GraphDSL, RunnableGraph, Sink}
+import akka.{Done, NotUsed}
 import com.softwaremill.kmq._
 import com.typesafe.scalalogging.StrictLogging
 
 import java.time.Clock
-import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.ExecutionContext
 
 /**
  * Combines functionality of [[RedeliverySink]] and [[CommitMarkerSink]].
  */
 class RedeliveryTrackerStream(markerConsumerSettings: ConsumerSettings[MarkerKey, MarkerValue],
-                              markersTopic: String, maxPartitions: Int)
+                              markersTopic: String)
                              (implicit system: ActorSystem, ec: ExecutionContext,
                               kafkaClients: KafkaClients, kmqConfig: KmqConfig, clock: Clock) extends StrictLogging {
 
   def run(): DrainingControl[Done] = {
     Consumer.committablePartitionedSource(markerConsumerSettings, Subscriptions.topics(markersTopic))
-      .mapAsyncUnordered(maxPartitions) {
+      .map {
```

Comment:
can you elaborate a bit what was the problem here?

I think this change is incorrect: we'll only ever handle one topic-parition,  as the stream handling that partition's messages will never end (the future returned by `run` will never be complete). So I'd say the previous version was correct

Reply:
Okay, I'm bringing back the mapAsync version. I still need to sort out the problem with broken DrainControl channel.